app/code/core/Mage/Sales/Model/Order/Creditmemo.php

```diff
@@ -324,9 +324,25 @@ public function addItem(Mage_Sales_Model_Order_Creditmemo_Item $item)
      *
      * @return Mage_Sales_Model_Order_Creditmemo
      */
-    public function collectTotals()
-    {
-        foreach ($this->getConfig()->getTotalModels() as $model) {
+     public function collectTotals()
+    {
+        $models = $this->getConfig()->getTotalModels();
+        $res = array();
+        $i=0;
+        $discount = 0;
+        foreach ($models as $name =>$model) {
+            if ($name=='discount') $discount = $i;
```

Comment:
whitespaces around == needed, and please wrap `$discount = $i;` with brackets `{}`